lib/rest-adapter.js

```diff
@@ -571,16 +577,16 @@ RestAdapter.prototype.allRoutes = function() {
 };
 
 RestAdapter.prototype.getClasses = function() {
-  return this.remotes.classes(this.options).map(function(c) {
-    return new RestClass(c);
+  return this.remotes.classes(this.options).map(c => {
+    return new RestClass(c, this.options);
```

Comment:
We must not store `RestAdapter` options inside `SharedClass` instance, because that breaks to logic merging class-level with app/adapter-level settings. Instead, REST-related options should be stored inside `RestClass`.